sklearn/tree/_tree.pyx

```diff
@@ -208,21 +212,19 @@ cdef class DepthFirstTreeBuilder(TreeBuilder):
                 parent = stack_record.parent
                 is_left = stack_record.is_left
                 impurity = stack_record.impurity
+                weighted_n_node_samples = stack_record.weighted_n_node_samples
+                sum_total = stack_record.sum_total
+                sq_sum_total = stack_record.sq_sum_total
                 n_constant_features = stack_record.n_constant_features
 
                 n_node_samples = end - start
-                splitter.node_reset(start, end, &weighted_n_node_samples)
+                splitter.node_reset(start, end, weighted_n_node_samples, sum_total, sq_sum_total)
```

Comment:
Seems reasonable. Initially I wanted to keep the code structure as similar as possible for this PR.